cluster-autoscaler/provisioningrequest/checkcapacity/provisioningclass.go

```diff
@@ -229,19 +232,8 @@ type combinedStatusSet struct {
 
 // Add adds a ScaleUpStatus to the combinedStatusSet.
 func (c *combinedStatusSet) Add(newStatus *status.ScaleUpStatus) {
-	// This represents the priority of the ScaleUpResult. The final result is the one with the highest priority in the set.
-	resultPriority := map[status.ScaleUpResult]int{
-		status.ScaleUpNotTried:           0,
-		status.ScaleUpNoOptionsAvailable: 1,
-		status.ScaleUpError:              2,
-		status.ScaleUpSuccessful:         3,
-	}
-
-	// If even one scaleUpSuccessful is present, the final result is ScaleUpSuccessful.
-	// If no ScaleUpSuccessful is present, and even one ScaleUpError is present, the final result is ScaleUpError.
-	// If no ScaleUpSuccessful or ScaleUpError is present, and even one ScaleUpNoOptionsAvailable is present, the final result is ScaleUpNoOptionsAvailable.
-	// If no ScaleUpSuccessful, ScaleUpError or ScaleUpNoOptionsAvailable is present, the final result is ScaleUpNotTried.
-	if resultPriority[c.Result] < resultPriority[newStatus.Result] {
+	// we only report the best result.
+	if newStatus.Result < c.Result {
```

Comment:
I'm really uncomfortable with the idea of relying on the coincidential order of statuses. If a new one needs to be added, there's no guarantee it'll be added in a way preserving the order we happen to prefer for error reporting.

`resultPriority` map might not be the nicest way to go about it, but I think it's more reliable as it's explicit. I realize there's a similar problem with this - any new status ending up as '0' and therefore the most preferred - but this can probably be covered by a unit test (to be added later) ensuring all statuses are covered.

Do you think we can skip this refactor?

Reply:
Yup, I'll revert this.